R/codes.R

```diff
@@ -0,0 +1,174 @@
+#' Look up descriptions for clinical codes
+#'
+#' Returns a data frame including descriptions for the codes of interest
+#'
+#' @param codes character. Vector of codes to lookup. If passing `"all"`, returns all codes.
+#' @param code_type character. Type of clinical code system to be searched.
+#'   Depends on what is available in the lookup tables. See [add_lookup_table()]
+#'   on how to add new lookup tables. This can also be configured through the `codeminer.code_type` option.
+#' @param preferred_description_only logical. If `TRUE`, only returns the preferred description for each code.
+#'   Default: `FALSE`.
+#'
+#' @return A `data.frame` containing the codes and their descriptions
+#' @export
+#' @family Clinical code lookups and mappings
+#' @seealso [add_lookup_table()] for adding new lookup tables to the database.
+#' @examples
+#' # Set up a temporary dummy database
+#' temp_db <- tempfile(fileext = ".duckdb")
+#' create_dummy_database(temp_db)
+#'
+#' # look up ICD10 codes
+#' CODES(
+#'   codes = c("E10", "E11"),
+#'   code_type = "icd10"
+#' )
+CODES <- function(
+  codes,
+  code_type = getOption("codeminer.code_type"),
+  version = "latest",
+  preferred_description_only = TRUE
+) {
+  check_lookup_args(code_type, version)
+
+  con <- connect_to_db()
+  check_database(con)
+
+  lookup_table <- get_lookup_table(con, code_type, version)
+
+  if (length(codes) == 1 && codes == "all") {
```

Comment:
Might need a validation helper e.g. `check_codes()`? Otherwise if user supplies something invalid like a list, then an odd error message may get raised later e.g. 

```r
> CODES(list("E10"), "icd10")
ℹ Using 'v0' as latest version
Error in `dplyr::filter()` at codeminer/R/codes.R:43:3:
ℹ In argument: `.data[["code"]] %in% list("E10")`
Caused by error in `FUN()`:
! Unknown input type: list
Run `rlang::last_trace()` to see where the error occurred.
```

Suggest using either `rlang::is_character()` or `rlang::is_atomic()`

Also, could use `identical()` here:

```suggestion
  if (identical(codes, "all")) {
```